ui/app/src/components/Navigation/PathNavigator/Widget.tsx

```diff
@@ -542,10 +545,16 @@ export default function (props: WidgetProps) {
   const [newFileDialog, setNewFileDialog] = useState(null);
   const [uploadDialog, setUploadDialog] = useState(null);
 
+  useEffect(() => {
+    dispatch(fetchSupportedLocales());
```

Comment:
Not checking if locales are fetched already.
Remove this call and add conditional fetch to the useSiteLocales hook